### DISCUSSION TITLE
Thermostat over climate, cannot use climate device's current temperature as temperature sensor

### DISCUSSION THREAD

**Discussion by jezzaaa** [jmcollin78/versatile_thermostat#441] [Q&A]
When I create a new "Thermostat over a climate" device/service, I must choose a temperature sensor. My real thermostat (ZigBee controller for a gas-powered heater) is entity climate.heater. The thermostat has a temperature sensor, reported as attribute "current-temperature" in the climate entity. I'm unable to select this as the temperature sensor.

When selecting the source for "Room Temperature" I would like to be able to choose the underlying climate device and have Versatile Thermostat use the number from the current_temperature attribute. Ideally, for a thermostat-over-client device, the underlying climate device would be the default source for room temperature.

**Comment by jmcollin78** [comment]
Hello @jezzaaa ,

I will transform this into a discussion, as it is really not an issue but more a question.

**Comment by jmcollin78** [comment]
1. using the internal device temp as room temperature sensor for a thermostat over climate is really a very bad idea. Internal device temp are known to be wrong because too near the heating device.
2. What you want to achieve is to have the room temperature reach the target and not the internal device temperature.

So in my opinion, you should have a temperature sensor in the room at the correct place and VTherm will regulate correctly.

If you want to use the internal temperature, then you have to build a template sensor with the internal temp value and use it as in the VTherm configuration. But really I want to discourage you for doing this.

**Reply by jezzaaa** [reply]
Thanks for considering my query.

I recognise that it's much better to use a separate temp sensor device to the heater.

However, in my case (and many similar heater's installed in my region) the heater's temperature sensor is incorporated into the control panel, and the control panel is installed onto a wall in a suitable place for measuring the comfort level. In some installations, the sensor/panel is installed close to the heater return vent so that it measures the temperature close to the end of the circuit of airflow.

So in these cases, the climate device's current temperature sensor is in the optimal location already, and adding another temp sensor in the same place would be redundant. And I'd prefer to not have to purchase a new device that gives the same numbers as a device I already use.

I think it would be a useful feature for many, especially for people who aren't familiar with creating sensors in HA. But I will give your suggestion a try. Thankyou.

**Reply by jmcollin78** [reply]
Ok, I understand. I that case, you can use the temperature of your device. And so you will have to extract this temperature as a sensor from your climate entity.
